packages/react/src/components/Storage/FileUploader/Previewer/index.tsx

```diff
@@ -35,22 +40,34 @@ export function Previewer({
             {translate('Drop files here or')}
           </Text>
           <UploadButton
-            {...rest}
+            acceptedFileTypes={acceptedFileTypes}
+            multiple={multiple}
+            onFileChange={onFileChange}
             className={'amplify-fileuploader__dropzone__button'}
           />
         </UploadDropZone>
-        <Text fontWeight="bold">{files.length} files selected</Text>
+        <Text fontWeight="bold">
+          {files.length} {translate('files selected')}
+        </Text>
         {files?.map((file, index) => (
-          <Tracker key={index}>{file.name}</Tracker>
+          <Tracker
+            file={file}
+            hasImage={file?.type.startsWith('image/')}
+            url={URL.createObjectURL(file)}
+            key={index}
+            onChange={(e): void => onNameChange(e, index)}
+            onCancel={() => onFileCancel(index)}
+            name={allFileNames[index]}
```

Comment:
Can we just use `files` to get the `name` instead of using `allFileNames`? 

We're maintaining two variables here (`allFileNames`, `files`), and IMO it'd be more maintainable to have one source of truth instead of two.

Reply:
Unfortunately, `files` is already being passed in as a prop. So I can't name it `files`. I could rename it to something else?

Reply:
Oh, I meant that we should not need `allFileNames` at all. Can we just remove `allFileNames`, and infer the file name directly from `files`?

Reply:
I can't. Because files, are an array of `File` and `File`  from a File Input is [readonly](https://stackoverflow.com/questions/55858559/change-the-filename-when-uploading-a-file). So I can't update the name in the `File` itself, without completely making a copy and replacing it. Not sure if that's the right approach for larger files. That's why I had to create a separate array with useState called `allFileNames` for just the name of files.